packages/dtslint/src/util.ts

```diff
@@ -7,6 +7,12 @@ export function packageNameFromPath(path: string): string {
   const base = basename(path);
   return /^v\d+(\.\d+)?$/.exec(base) || /^ts\d\.\d/.exec(base) ? basename(dirname(path)) : base;
 }
+export function packageDirectoryNameWithVersionFromPath(path: string): string {
```

Comment:
No actual change necessary, but I do hope that one day we can eliminate all of the path-dependent behavior and just use package.json as the source of truth; that's all package linking cares about.

Reply:
I don’t think that’s possible for this—exemptions for mismatched npm versions need to be specified at the `react/v16` level, but there’s nothing in `react/v16/package.json` that would tell us, in isolation, whether we should refer to it as `react` or `react/v16`

Reply:
There is; the name + the version. Post-pnpm, if it weren't for DT tooling, the packages could be placed in any directory at all. The name + version combo in package.json is all that is used to link them together.

Reply:
Yeah, I have a minor feeling that the idea of a "latest" package is generally no longer needed in a post-monorepo world, but, I'm just musing and I still don't want you to do anything here 😄